raiden/network/rpc/client.py

```diff
@@ -686,7 +686,17 @@ def eth_estimateGas(
             startgas,
             gasprice,
         )
-        res = self.call('eth_estimateGas', json_data)
+        try:
+            res = self.call('eth_estimateGas', json_data)
+        except EthNodeCommunicationError as e:
+            tx_would_fail = str(e) in [
```

Comment:
Please, use the error code instead of the error text

Reply:
I did exactly that in the beginning ... but then realized that it's not returned by tiny rpc :(
Unless I make a specific case inside `eth_call`, take the raw json of the reply and insert it in the exception

Reply:
The error code should be expose in the exception raised by tinyrpc:
https://github.com/mbr/tinyrpc/blob/master/tinyrpc/protocols/jsonrpc.py#L269

Reply:
@hackaugusto 

Yes it can be returned, but what I meant is we [don't currently return it](https://github.com/raiden-network/raiden/blob/41b1fa41a606fb27fe2e3272ab3cba17ded4fcc4/raiden/network/rpc/client.py#L499) via the EthNodeCommunicationError. And the string is used at some point.

I see two options. 

1. Add the fields to the `EthNodeCommunicationError` and have both an error message and an error code in there. This would be a bit cleaner and allow for error messages to be cleanly shown where we turn `EthNodeCommunicationError` into a `str while still being able to extract the error code where needed.
2. Just append the error code to the already returned message and somehow parse it off. I don't like this approach.

Any other approach? I would go with 1.

Reply:
Go with 1, using the error message for anything other than a human readable message is a why to ask for failures in the future.

Reply:
agreed